app/src/org/commcare/activities/EntityDetailActivity.java

```diff
@@ -253,7 +253,7 @@ public boolean onCreateOptionsMenu(Menu menu) {
     @Override
     public boolean onPrepareOptionsMenu(Menu menu) {
         super.onPrepareOptionsMenu(menu);
-        menu.findItem(MENU_PRINT_DETAIL).setVisible(detail.isPrintEnabled());
+        menu.findItem(MENU_PRINT_DETAIL).setVisible(detail != null && detail.isPrintEnabled());

```

Comment:
Agree that not a good solution. I don't really have a clear reason on why it might be happening and was not able to repro it. But thinks it might be because of `onPrepareOptionsMenu` getting called out of order for some rare scenarios since I don't see similar crashes for detail being null elsewhere in this activity.

Reply:
Got it. totally fine to just move forward with the != check then